### ISSUE TITLE
Theme system rework

### ISSUE THREAD

**Issue by tenphi** [tenphi/nude#6]
Theme system have to be changed and simplified. Current implementation doesn't meet contrast ratio WCAG requirements. It also requires to manually define basic themes and unable to adaptively change shadow intensity in complex cases.

New theme spec can solve all this problems. 

In common scenarios the user only need to specify single brand color:

```
<nu-theme from="#1885d9"></nu-theme>
```

It would be enough in 95-99% cases because it will generate a set of themes for multiple purposes.
